packages/ado-extension/src/ado-extension.ts

```diff
@@ -17,9 +18,14 @@ export function runScan(): void {
         await logger.setup();
 
         const scanner = container.get(Scanner);
-        process.exit((await scanner.scan()) ? ExitCode.ScanCompletedNoUserActionIsNeeded : ExitCode.ScanCompletedUserActionIsNeeded);
-    })().catch((error) => {
-        console.log('##[error][Exception] Exception thrown in extension: ', error);
-        process.exit(ExitCode.ScanFailedToComplete);
+        const taskSucceeded = await scanner.scan();
+
+        if (taskSucceeded) {
+            adoTask.setResult(adoTask.TaskResult.Succeeded, 'Scan completed successfully');
+        } else {
+            adoTask.setResult(adoTask.TaskResult.Failed, logger.getAllErrors() || 'Scan failed');
+        }
```

Comment:
The success message is never shown in the logs, summary, or email notification. I think we can simplify to:

```suggestion
        if (taskSucceeded) {
            adoTask.setResult(adoTask.TaskResult.Succeeded);
        } else {
            adoTask.setResult(adoTask.TaskResult.Failed, logger.getAllErrors() || 'Scan failed');
        }
```

Reply:
According to the documentation and after looking at a lot of other extension codebases, setting a successful result is not required and not often implemented:

> Sets the result of the task. Execution will continue. If not set, task will be Succeeded. If multiple calls are made to setResult the most pessimistic call wins (Failed) regardless of the order of calls.

I think we can change this to only setResult if `!taskSucceeded`.